### PR TITLE
Revise to emphasize stability and uniqueness of ids

### PR DESCRIPTION
**Pull request by bdarcus**
This changes the language on id URIs to be more assertive, and remove
the "dereferencable" goal.

Also includes a suggestion to consider urn:uuid URIs to ensure stability
and uniqueness.

### CODE REVIEW

File: specification.rst [hunk 1]
```diff
@@ -179,9 +179,10 @@ In independent styles, ``cs:info`` has the following child elements:
     indicates the field(s) for which the style is relevant.
 
 ``cs:id``
-    Must appear once. The element should contain a URI to establish the identity
-    of the style. A stable, unique and dereferenceable URI is desired for
-    publicly available styles.
+    Must appear once, and contain a URI to establish the identity
+    of the style. For publicly available styles, this URI must be stable and
```
Comment (bdarcus) [inline specification.rst:183]: "must" is strong language in a spec, and probably not appropriate here; "should" could be coupled with a following sentence that explains why?

**Comment by bdarcus** [comment]
@dstillman - does this address the problem noted in your proposal from awhile back? 

https://discourse.citationstyles.org/t/proposal-make-style-ids-immutable

Could be coupled with stronger requirements on the styles repo, say that new styles MUST use urn:uuids.

**Comment by dstillman** [comment]
I wouldn't even call it a URI — that's going to cause people to continue to use URL-style ids. And in my view, the `urn:` prefix is pointless and adds absolutely nothing of value.

I would just call this a stable identifier (maybe define valid characters?) and recommend using a UUID directly.

(This addresses the primary issue from that thread, but there are others related to other fields.)

**Comment by bdarcus** [comment]
So what should the data type be for id?

It's been a URI since the beginning.

Essentially you're arguing for broader, but what?

**Comment by dstillman** [comment]
Just a string. Processors should probably be prepared to handle at least `[A-Za-z0-9/:._-]` to account for existing URIs, but there's no reason new ones shouldn't be UUIDs.

**Comment by bdarcus** [comment]
OK. But as I said, what's in the spec is a different question from what rules we adopt to accept styles to the repo.

But assuming we adopt your suggestion, how's this for spec language:

> Must appear once, and contain a stable identifier to establish the identity of the style. For publicly available styles, this identifier must be globally unique, and conform to the following regular expression: `[A-Za-z0-9\/:._-]`. One may use a URL, but we recommend a UUID for new styles.

Is this a preferable solution to adding a new `cs:uuid` element, or (another option) adding a type attribute on the id element, like:

``` xml
<id type="uuid">...</id>
```

**Comment by dstillman** [comment]
I'm not really sure what the goal is here. Are you being prescriptive or defining the world as it is?

As far as I'm concerned, we should just say that the id for any new style should be a UUID. It's not about "publicly available styles" — an id that conflicts with another local style or that is changed could break things locally too. There's no reason to say that one "may use a URL", which might encourage deferenceable URLs that would be tempting to change later. One could use all Unicode astral plane characters and it might work, but that doesn't need to be in the spec.

Processors and applications need to know that they may encounter existing styles that use URIs — we could just say URIs, since that was in the spec, or we could be realists and retroactively limit it to `[A-Za-z0-9\/:._-]` — but that's separate from what the spec should require for anyone creating a new valid style.

**Comment by dstillman** [comment]
E.g.,

> Must appear once and contain a stable, unique identifier to establish the identity of the style. For historical reasons, existing styles may use URIs, but new styles should use a UUID to guarantee stability and uniqueness.

**Comment by bdarcus** [comment]
I committed your suggested language. Thanks.

I asked about the datatype because I was assuming your suggestion would require a change in the schema.

Current schema is:

```
element cs:id { xsd:anyURI }
```

But when I tested it just now, both jing and rnv see a UUID as valid.

@adam3smith and @rmzelle - any thoughts at this point?

If we want to merge this change, I can open a separate issue on the schema repo to settle what, if anything, we should do with `cs:id` there.

**Pull request merged**